pkg/loadbalancerservice/router.go

```diff
@@ -0,0 +1,135 @@
+package loadbalancerservice
+
+import (
+	"slices"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/vishvananda/netlink"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultRouterServiceName            = "router-default"
+	defaultRouterServiceNamespace       = "openshift-ingress"
+	defaultRouterServiceAnnotationKey   = "ingresscontroller.operator.openshift.io/owning-ingresscontroller"
+	defaultRouterServiceAnnotationValue = "default"
+)
+
+type serviceUpdateFunction func([]string) error
+
+func defaultRouterWatch(ipAddresses, nicNames []string, updateFunc serviceUpdateFunction, stopCh <-chan struct{}) {
+	for {
+		ips, err := defaultRouterListenAddresses(ipAddresses, nicNames)
+		if err != nil {
+			klog.Errorf("unable to determine default router listening addresses: %v", err)
+			continue
+		}
+		if err := updateFunc(ips); err != nil {
+			klog.Errorf("unable to update default router service status: %v", err)
+			continue
+		}
+		break
+	}
+
+	updateChan := make(chan netlink.AddrUpdate)
+	doneChan := make(chan struct{})
+	err := netlink.AddrSubscribe(updateChan, doneChan)
+	if err != nil {
+		klog.Errorf("unable to subscribe to IP address changes: %v", err)
+		return
```

Comment:
Is this the best we can do? It's not progagated so MicroShift could continue running without this functionality and I'm not sure if that's a good thing

Reply:
See next comment.